modules/core/04-channel/types/upgrade.go

```diff
@@ -83,13 +83,21 @@ func (u *UpgradeError) Error() string {
 	return u.err.Error()
 }
 
-// Is returns true if the underlying error is of the given err type.
-func (u *UpgradeError) Is(err error) bool {
-	return errors.Is(u.err, err)
+// Is returns true if the of the provided error is an upgrade error.
+func (*UpgradeError) Is(err error) bool {
+	_, ok := err.(*UpgradeError)
```

Comment:
I think this is the simplest approach, by implementing it this way, when we call `errors.Is` with an `UpgradeError` as an argument, it will handle the traversal of the error stack and eventually hit this.